_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,221 @@
+---
+id: j7zzqr
+name: ARIA role is permitted for the element.
+rule_type: atomic
+description: |
+  This rule checks that explicit WAI-ARIA roles are allowed for the element they are specified on.
+accessibility_requirements:
+  html-aria:docconformance:
+    title: ARIA in HTML, 4. Document conformance requirements for use of ARIA attributes in HTML
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any [HTML element][namespaced element] which has an [explicit semantic role][explicit role].
+
+## Expectation
+
+For each test target, its [explicit semantic role][explicit role] is allowed on this element, according to the [ARIA in HTML specification][aria in html document conformance].
+
+## Assumptions
+
+There are no assumptions.
+
+## Accessibility Support
+
+There are no accessibility support issues known.
+
+## Background
+
+Each element also has an [implicit semantic role][implicit role] defined in the [HTML Accessibility API Mappings][html aam:roles]. Setting the [explicit role][] as the same as the [implicit role][] is not recommended but nonetheless allowed. This rule does not check that scenario.
+
+This rule does not check specifically whether the [explicit role][] is deprecated, or whether it is the special `generic` role that should not be used by authors. These are not conformance requirement, or are checked by other rules.
+
+This rules apply to every element, even if they are not [included in the accessibility tree][]. This is because the roles of `none` or `presentation` are only allowed on certain elements. If the rule was only looking at content [included in the accessibility tree][], it wouldn't flag incorrect use of these roles.
```

Comment:
```suggestion
This rule applies to every element, even if they are not [included in the accessibility tree][]. This is because the roles of `none` or `presentation` are only allowed on certain elements. If the rule was only looking at content [included in the accessibility tree][], it wouldn't flag incorrect use of these roles.
```

this section should probably also mention content hidden from the a11y tree with aria-hidden=true, or other host language features (per the example on line 182)

Reply:
@giacomo-petri Can you get a look at it now? I tried to improve the situation…